File: src/Maruko/Application/Servers/Dto/PageDto.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Maruko.Core.Application.Servers.Dto
{
    public abstract class PageDto
    {
        /// <summary>
        /// 动态查询条件
        /// </summary>
        public List<DynamicFilter> DynamicFilters { get; set; }

        /// <summary>
        /// 当前页
        /// </summary>
        [Range(1, 2147483647, ErrorMessage = "PageIndex的值在1~2147483647之间")]
        public int PageIndex { get; set; } = 1;

        /// <summary>
        /// 一页数据总数
        /// </summary>
        [Range(1, 2147483647, ErrorMessage = "PageSize的值在1~2147483647之间")]
        public int PageMax { get; set; } = 20;
    }
}

File: act/act-framework/act-framework-core/src/main/java/com/act/core/application/PageDto.java
package com.act.core.application;

import lombok.Data;

import java.sql.Wrapper;
import java.util.ArrayList;
import java.util.List;

@Data
public class PageDto {
    private List<DynamicFilter> dynamicFilters;
    private Integer pageIndex = 1;
    private Integer pageSize = 20;

    public PageDto() {
        dynamicFilters = new ArrayList<>();
    }
}

